skyd/api.go

```diff
@@ -68,44 +103,66 @@ func NewAPI(nginxHost string, nginxPort int, skydHost, skydPassword string, skyd
 	}, nil
 }
 
-// BlockSkylinks will perform an API call to skyd to block the given skylinks
-func (api *api) BlockHashes(hashes []string) error {
-	// Build the call to skyd.
-	reqBody := skyapi.SkynetBlocklistPOST{
-		Add:    hashes,
+// BlockHashes will perform an API call to skyd to block the given hashes. It
+// returns which hashes were blocked, which hashes were invalid and potentially
+// an error.
+func (api *api) BlockHashes(hashes []database.Hash) ([]database.Hash, []database.Hash, error) {
+	api.staticLogger.Debugf("blocking %v hashes", len(hashes))
+
+	// convert the hashes to strings
```

Comment:
do you need this? Aren't the hashes marshalled to strings by `json.Marshal` already? Their `json.Marshaler` implementation might even use the `.String()` method.

Reply:
I think I do need that because `SkynetBlocklistPOST` expects a `[]string`...

```
	SkynetBlocklistPOST struct {
		Add    []string `json:"add"`
		Remove []string `json:"remove"`

		// IsHash indicates if the supplied Add and Remove strings are already
		// hashes of Skylinks
		IsHash bool `json:"ishash"`
	}
```